app/models/operatingsystem.rb

```diff
@@ -251,7 +251,7 @@ def boot_filename(host = nil)
     return default_boot_filename if host.nil? || host.pxe_loader.nil?
     return host.foreman_url('iPXE') if host.pxe_loader == 'iPXE Embedded'
     architecture = host.arch.nil? ? '' : host.arch.bootfilename_efi
-    boot_uri = URI.parse(host.subnet.httpboot? ? host.subnet.httpboot.url : Setting[:unattended_url])
+    boot_uri = URI.parse((host.subnet.present? && host.subnet.httpboot?) ? host.subnet.httpboot.url : Setting[:unattended_url])
```

Comment:
not crucial, but you can use the safe navigation operator here
```suggestion
    boot_uri = URI.parse((host.subnet&.httpboot?) ? host.subnet.httpboot.url : Setting[:unattended_url])
```